File: Frontend/src/modules/admin/feature/expiryAlert/services/expiry-alert.service.ts
import api from "@/common/api";

import type {
  ExpiryAlertListParams,
  ExpiryAlertListResponse,
  ExpiryAlertSummaryResponse,
  InventoryLotDetailResponse,
  PageResponse,
} from "../types/expiry-alert.type";

const cleanParams = (
  params: Record<string, unknown>
): Record<string, unknown> => {
  return Object.fromEntries(
    Object.entries(params).filter(
      ([, value]) => value !== undefined && value !== null && value !== ""
    )
  );
};

const expiryAlertService = {
  /**
   * GET /api/admin/expiry-alerts
   */
  async getList(
    params: ExpiryAlertListParams
  ): Promise<PageResponse<ExpiryAlertListResponse>> {
    const response = await api.get("/admin/expiry-alerts", {
      params: cleanParams({
        group: params.group ?? "NEAR_EXPIRY",

        keyword: params.keyword?.trim() || undefined,

        fromDays: params.fromDays ?? undefined,

        toDays: params.toDays ?? undefined,

        page: params.page ?? 0,

        size: params.size ?? 20,
      }),
    });

    return response.data;
  },

  /**
   * GET /api/admin/expiry-alerts/summary
   */
  async getSummary(): Promise<ExpiryAlertSummaryResponse> {
    const response = await api.get("/admin/expiry-alerts/summary");

    return response.data;
  },

  /**
   * GET /api/admin/expiry-alerts/{id}
   */
  async getDetail(id: number): Promise<InventoryLotDetailResponse> {
    const response = await api.get(`/admin/expiry-alerts/${id}`);

    return response.data;
  },
};

export default expiryAlertService;

File: Backend/src/main/java/org/example/datn_sd69/modules/expiryalert/service/ExpiryAlertService.java
package org.example.datn_sd69.modules.expiryalert.service;

import org.example.datn_sd69.enums.ExpiryAlertGroup;
import org.example.datn_sd69.modules.expiryalert.dto.response.ExpiryAlertListResponse;
import org.example.datn_sd69.modules.expiryalert.dto.response.ExpiryAlertSummaryResponse;
import org.example.datn_sd69.modules.inventorylot.dto.response.InventoryLotDetailResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ExpiryAlertService {

    Page<ExpiryAlertListResponse> getList(
            ExpiryAlertGroup group,
            String keyword,
            Integer fromDays,
            Integer toDays,
            Pageable pageable
    );

    ExpiryAlertSummaryResponse getSummary();

    InventoryLotDetailResponse getDetail(Integer id);
}